com.unity.netcode.gameobjects/Runtime/Messaging/Messages/ConnectionApprovedMessage.cs

```diff
@@ -315,14 +319,14 @@ public void Handle(ref NetworkContext context)
                 if (networkManager.DistributedAuthorityMode && networkManager.CMBServiceConnection && networkManager.LocalClient.IsSessionOwner && networkManager.NetworkConfig.EnableSceneManagement)
                 {
                     // Mark the client being connected
-                    networkManager.IsConnectedClient = true;
+                    networkManager.IsConnectedClient = networkManager.ConnectionManager.LocalClient.IsApproved;
```

Comment:
Will this update when the client is approved if they're not approved at this point?

Reply:
This is actually set within this script (`ConnectionApprovedMessage`).